dataflow/src/node/process.rs

```diff
@@ -81,7 +81,15 @@ impl Node {
                         // we need to own the data
                         let old_data = mem::replace(data, Records::default());
 
-                        match i.on_input_raw(from, old_data, &mut tracer, &replay, nodes, state) {
+                        match i.on_input_raw(
+                            from,
+                            old_data,
+                            &mut tracer,
+                            &replay,
+                            nodes,
+                            state,
+                            on_shard,
```

Comment:
This makes me a little sad, and makes me even more inclined to move `Base` out of `ops`. Unfortunately it'd be pretty inefficient for the shard to be set anywhere further up (e.g., in `node/process`), so I think we're stuck doing things this way for now.